### ISSUE TITLE
Error 0x80070057

### ISSUE THREAD

**Issue by PoolloverNathan** [microsoft/WSL#4800]
I get error 0x80070057 (#7) when I am trying to change the ubuntu1804 default user, and starting it fails without any output. However, when I explicitly tell wsl.exe to use a non-default user, it works. Why can't I change the default user? This started happening after I restarted Explorer a couple of minutes ago to clear out my old windows.

**Comment by robertoandrade** [comment]
Having the same issue with WSL2 on Windows 10 Pro update 2004 build 19631 and Ubuntu 20.04. Can't get back into the installation after running `ubuntu config --default-user root`, not even resetting it back to the old user allows be to reenter WSL.

**Comment by monkeyWie** [comment]
I have this problem too
![image](https://user-images.githubusercontent.com/13160176/84345857-136b1e80-abe1-11ea-959a-8f9bfd79c7d2.png)

**Comment by therealkenc** [comment]
[Detailed logs](https://github.com/microsoft/WSL/blob/master/CONTRIBUTING.md#8-detailed-logs).

**Comment by alexsuslin** [comment]
> I get error 0x80070057 (#7) when I am trying to change the ubuntu1804 default user, and starting it fails without any output. However, when I explicitly tell wsl.exe to use a non-default user, it works. Why can't I change the default user? This started happening after I restarted Explorer a couple of minutes ago to clear out my old windows.

The same for ubuntu2004.exe config --default-user root
`Error: 0x80070057 The parameter is incorrect.`

**Comment by findli** [comment]
lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.6 LTS
Release:        16.04
Codename:       xenial

**Comment by wshon** [comment]
The same for Debian

![图片](https://user-images.githubusercontent.com/16683806/131777567-71d0f164-ef57-41f7-9f23-26b1e66bc1e0.png)

![图片](https://user-images.githubusercontent.com/16683806/131777679-9c907147-0dfe-4e1c-be37-e3e17fd39c6e.png)

**Comment by XHunter74** [comment]
Also had such an issue because switched to an user which shell had nologin value in passwd. Changed it to bash and the issue was gone.

**Comment by kiuyaku** [comment]
> 也有这样的问题，因为切换到 shell 在 passwd 中没有登录值的用户。将其更改为 bash，问题就消失了。

I used bash in windows, entered linux and modified the default bash of the corresponding user, the problem was solved!
Thanks a lot.

**Comment by XHunter74** [comment]
Easily: wsl -u root

**Comment by q00u** [comment]
> Easily: wsl -u root, for example

wsl is not working. If I try to enter wsl, it drops me back at the command line. @kiuyaku said "I used bash in windows", so I was asking if there's a way to modify the default bash of the corresponding user, _from Windows_, to fix the broken wsl.

**Comment by XHunter74** [comment]
wsl must work. btw bash in windows - I think it is cygwin
Also, you can try to run ssh from windows and use it.

**Comment by kiuyaku** [comment]
> > 轻松：例如 wsl -u root
> 
> wsl 不起作用。如果我尝试输入 wsl，它会让我回到命令行。@kiuyaku说“我在 Windows 中使用了 bash”，所以我问是否有办法_从 Windows_修改相应用户的默认 bash以修复损坏的 wsl。

When I first had a problem, I modified the WSL to enter the default root user. However, it seems that the root user has not set the default bash in the WSL system, so the windows error appears.

To solve the problem, I opened CMD in windbos and directly entered bash. CMD directly enters the WSL system. Then I modified the default bash of root / user, and then modified XXX config -- default user to solve this problem.
